lark/tree_templates.py

```diff
@@ -49,15 +53,21 @@ def _get_tree(self, template: TreeOrCode) -> Tree[str]:
             assert self._parse
             template = self._parse(template)
 
-        assert isinstance(template, Tree)
+        if not isinstance(template, Tree):
+            raise TypeError("template parser must return a Tree instance")
+
         return template
 
     def __call__(self, template: Tree[str]) -> 'Template':
         return Template(template, conf=self)
 
-    def _match_tree_template(self, template: TreeOrCode, tree: TreeOrCode) -> Optional[Dict[str, TreeOrCode]]:
+    def _match_tree_template(self, template: TreeOrCode, tree: Branch) -> Optional[MatchResult]:
+        """Returns dict of {var: match} if found a match, else None
+        """
         template_var = self.test_var(template)
         if template_var:
+            if not isinstance(tree, Tree):
+                raise TypeError(f"Template variables can only match Tree instances. Not {tree}")

```

Comment:
This should be either `type(tree)` or `{tree!r}`

Reply:
Fixed.